frontend/src/metabase/query_builder/selectors.js

```diff
@@ -314,6 +315,23 @@ export function normalizeQuery(query, tableMetadata) {
         );
       });
     }
+
+    // sort query.join[int].fields
+    if (query.query.joins) {
+      query = updateIn(query, ["query", "joins"], joins =>
+        joins.map(joinedTable => {
+          if (!joinedTable.fields || joinedTable.fields === "all") {
```

Comment:
1. The first predicate is to prevent [this test](https://github.com/metabase/metabase/runs/5751459337?check_suite_focus=true) from failing
2. I found the joined table's `fields` could be a string `"all"`

![image](https://user-images.githubusercontent.com/1937582/160796415-1981a92e-2649-4681-973f-f2f3ab520780.png)